### PR TITLE
Avoid grid re-renders when clicking the selected cell

### LINKED ISSUE #2388: Avoid grid rerender on same cell click multiple times
When the same cell is clicked again and again, the selected position state is getting update and which causes grid to get rerendered on each click. To avoid this a condition can be checked like this 

`function selectCell(position, enableEditor = false) {
    if (!isCellWithinBounds(position)) return;
    commitEditorChanges();

    if (enableEditor && isCellEditable(position)) {
      const row = rows[position.rowIdx];
      setSelectedPosition({ ...position,
        mode: 'EDIT',
        key: null,
        row,
        originalRow: row
      });
    } else if (position.idx !== selectedPosition.idx || position.rowIdx !== selectedPosition.rowIdx || selectedPosition.mode !== 'SELECT') {
      setSelectedPosition({ ...position,
        mode: 'SELECT'
      });
    }

    onSelectedCellChange == null ? void 0 : onSelectedCellChange({ ...position
    });
  }`

Please check the else if condition. Let me know your thoughts

### PR DESCRIPTION
**Pull request by nstepien**
Fixes #2388

### CODE REVIEW

File: src/DataGrid.tsx [hunk 1]
```diff
@@ -686,10 +686,16 @@ function DataGrid<R, SR, K extends Key>(
...
+      selectedPosition.mode !== 'SELECT' ||
+      selectedPosition.idx !== position.idx ||
+      selectedPosition.rowIdx !== position.rowIdx
+    ) {
+      // Avoid re-renders if the selected cell state is the same
+      // TODO: replace with a #record? https://github.com/microsoft/TypeScript/issues/39831
       setSelectedPosition({ ...position, mode: 'SELECT' });
     }
-    onSelectedCellChange?.({ ...position });
+    onSelectedCellChange?.(position);
```
Comment (amanmahajan7) [inline src/DataGrid.tsx:698]: Should we call `onSelectedCellChange` if position is not changed?

File: src/DataGrid.tsx [hunk 2]
```diff
@@ -686,10 +686,16 @@ function DataGrid<R, SR, K extends Key>(
     if (enableEditor && isCellEditable(position)) {
       const row = rows[position.rowIdx] as R;
       setSelectedPosition({ ...position, mode: 'EDIT', key: null, row, originalRow: row });
-    } else {
+    } else if (
+      selectedPosition.mode !== 'SELECT' ||
```
Comment (amanmahajan7) [inline src/DataGrid.tsx:690]: Can we add a test?

Comment (nstepien) [reply src/DataGrid.tsx:690]: How? 🤔

Comment (amanmahajan7) [reply src/DataGrid.tsx:690]: If we do not call `onSelectedCellChange` then we can use it to test this change

**Review by amanmahajan7 — APPROVED** [review]

**Pull request merged**